docs/How_To/Tutorials/add_citation.rst

```diff
@@ -0,0 +1,161 @@
+
+In this tutorial, we will add a citation to the Sherpa response. This is a great way to validate the response from Sherpa and to give credit to the source of the data.
+
+Add Citation to Google Search
+*****************************
+
+The built-in `GoogleSearch` action already handles citations internally, you only need to enable it using a `CitationValidation` in the validation step of the agent. 
+
+First add the citation validation in the configuration of the agent if it is not already there.:
+
+.. code-block:: yaml
+
+    citation_validation:  # The tool used to validate and add citation to the answer
+    _target_: sherpa_ai.output_parsers.citation_validation.CitationValidation
+    sequence_threshold: 0.6
+    jaccard_threshold: 0.6
+    token_overlap: 0.6
+
+Then, add the citation validation step to the validations available in the agent:
+
+.. code-block:: yaml
+
+    qa_agent:
+        ...
+        validations:
+            - ${citation_validation}
+
+.. note:: 
+    The ... in the configuration above represents the rest of the configuration of the agent.
+
+To test the citation validation, lets temporarily remove the document search action from the agent to ensure that the agent will always use the Google search action.
+
+.. code-block:: yaml
+
+    qa_agent:
+        ...
+        actions:
+            # - ${doc_search}
+            - ${google_search}
+
+Now, run the agent and ask a question that will trigger the Google search action. The agent should return the response with the citation.
+
+.. code-block:: bash
+
+    Ask me a question: What is data leakage in machine learning
+    2024-05-08 23:50:35.933 | INFO     | sherpa_ai.agents.base:run:70 - Action selected: ('Google Search', {'query': 'What is data leakage in machine learning'})
+    2024-05-08 23:50:37.528 | INFO     | sherpa_ai.agents.base:run:70 - Action selected: None
+    Data leakage (or leakage) in machine learning occurs when the training data includes information about the target that will not be available during prediction [1](https://machinelearningmastery.com/data-leakage-machine-learning/). This can result in high performance on the training set and validation data, but the model may perform poorly in production [1](https://machinelearningmastery.com/data-leakage-machine-learning/)
+
+
+Add Citation to Customized Actions
+**********************************
+
+The above example shows how to add citations to the Google search action. However, sometimes we may also want to add citations to the responses from the document search action. In this case, we need to manually add the citation to the response. 
+
+The `DocumentSearch` action inherit from the `BaseAction` class, which has a method `add_resources` that can be used to add a citation to the response. The `add_resources` method takes a list of dictionaries, each dictionary should contain the following keys:
+
+- `Document`: Content of the resource.
```

Comment:
is this specifically the chunk that was placed in the context? if so perhaps clarify that

Reply:
I added some description to it.